DesignSafe-CI/portal
# Task/DES-1789/1790 - Add "Natural Hazard Event" and "Field Research Type" fields to projects

## Overview: ##
1. Added Natural Hazard Event field from the FR model to all other project Models
2. Added Field Research Type field to the FR modl

## PR Status: ##


## Related Jira tickets: ##

* [DES-1789](https://jira.tacc.utexas.edu/browse/DES-1789)
* [DES-1790](https://jira.tacc.utexas.edu/browse/DES-1790)

## Summary of Changes: ##
1. Extended the existing field from Field Research projects called "Event Type" to all other project types. This is a required field in the Edit Project modal. For all projects except Field Research this field is called "Natural Hazard Type".
2. Added an additional field to the FR model called "Field Research Type" this is a required field for FR projects.

- Fixed some publication preview issues with file listing related to the Other project type (we're not listing files there so I removed it)
- Fixed some inconsistencies with adding/removing project members.

## Testing Steps: ##
1. Navigate to all project types (experimental, simulation, hybrid simulation, field research, and other) open the Edit Project modal.
2. From the modal you should see the newly added field "Natural Hazard Type" just under the "Project Type" field. Additionally the FR model will have a "Field Research Type" field just below the "Natural Hazard Type" field.
3. Try to add/remove save inputs in these fields as needed
4. You should be able to see the updates on the project in the red highlighted areas in the attached images...

## UI Photos:
**FIELD RESEARCH - EDIT PROJECT MODAL**
<img width="726" alt="fr_edit_modal" src="https://user-images.githubusercontent.com/29575979/102925666-7162a380-4459-11eb-8184-c2e630b1ff04.png">

**EXP, SIM, HYBSIM, OTHER - EDIT PROJECT MODAL**
<img width="722" alt="project_edit_modal" src="https://user-images.githubusercontent.com/29575979/102925723-8c351800-4459-11eb-9ef9-a59d3cf26982.png">

**FIELD RESEARCH - PROJECT/PUBLICATION VIEWS**
<img width="779" alt="fr_meta" src="https://user-images.githubusercontent.com/29575979/102925777-a40c9c00-4459-11eb-85a9-487e0b220c72.png">

**EXP, SIM, HYBSIM, OTHER - PROJECT/PUBLICATION VIEWS**
<img width="766" alt="project_meta" src="https://user-images.githubusercontent.com/29575979/102925827-b71f6c00-4459-11eb-9062-8068ac59df40.png">
